test/integration_test/kubevirt_test.go

```diff
@@ -0,0 +1,161 @@
+	require.NoError(t, err, "Error creating template")
+
+	t.Run("kubevirtDeployFedoraVMWithClonePVC", kubevirtDeployFedoraVMWithClonePVC)
+}
+
+func kubevirtDeployFedoraVMWithClonePVC(t *testing.T) {
+	var testrailID, testResult = 50803, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult)
+	defer updateTestRail(&testResult, testrailID, runID)
+	instanceID := "vm"
+	appKey := "kubevirt-fedora"
+	deployedVMName := "test-vm-csi"
+
+	kubevirtVMDeployAndValidate(
+		t,
+		instanceID,
+		appKey,
+		deployedVMName,
+	)
+
+	// If we are here then the test has passed
+	testResult = testResultPass
+	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func kubevirtVMDeployAndValidate(
+	t *testing.T,
+	instanceID string,
+	appKey string,
+	deployedVMName string,
+) []*scheduler.Context {
+	ctxs, err := schedulerDriver.Schedule(instanceID,
+		scheduler.ScheduleOptions{
+			AppKeys: []string{appKey},
+		})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for app to get to running state")
+
+	namespace := appKey + "-" + instanceID
+	vms, err := kubevirt.Instance().ListVirtualMachines(namespace)
+	require.NoError(t, err, "Error listing virtual machines")
+
+	for _, vm := range vms.Items {
+		validateVM(t, vm.Name, vm.Namespace, deployedVMName)
+	}
+	return ctxs
+}
+
+func validateVM(t *testing.T, name, namespace, vmName string) {
+	currentVM, err := kubevirt.Instance().GetVirtualMachine(name, namespace)
+	require.NoError(t, err, "error in getting deployed VM in namespace: %s", namespace)
+	require.Equal(t, currentVM.Name, vmName, "VM %s has not been deployed in namespace: %s", vmName, namespace)
+	require.Equal(t, currentVM.Status.Created, true, "VM %s created status is: %t", currentVM.Name, currentVM.Status.Created)
+	require.Equal(t, currentVM.Status.Ready, true, "VM %s ready status is: %t", currentVM.Name, currentVM.Status.Ready)
+
+	// TODO add more validations here if required
+}
+
+func createTemplatePVC() error {
+	_, err := schedulerDriver.Schedule("",
+		scheduler.ScheduleOptions{
+			AppKeys:   []string{"kubevirt-templates"},
+			Namespace: "openshift-virtualization-os-images",
+		})
+	if err != nil {
+		return fmt.Errorf("error deploying kubevirt templates")
+	}
+	// TODO: wait for imported pod to complete
+	var importerPod *corev1.Pod
+	waitForImporterPodStart := func() (interface{}, bool, error) {
+		allPods, err := core.Instance().GetPods("openshift-virtualization-os-images", nil)
+		if err != nil || len(allPods.Items) == 0 {
+			return nil, true, fmt.Errorf("No importer pod not started yet. Retrying")
+		}
+		for _, pod := range allPods.Items {
+			if strings.Contains(pod.Name, importerPodPrefix) {
+				importerPod, err = core.Instance().GetPodByName(pod.Name, "openshift-virtualization-os-images")
```

Comment:
why is GetPodByName() needed? We have just called GetPods() in the same function. Why can't we use the pod in the allPods.Items?